analysis-tools/multiple_runs_comparison.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.pyplot import axis
+import pandas as pd
+from pandas import DataFrame
+
+
+@dataclass
+class Simulation:
+    path: Path
+    prefix: str
+    results: List[DataFrame] = field(init=False)
+
+    def __post_init__(self):
+        # fills results
+        folders = [self.path / folder_name for folder_name in os.listdir(self.path)
+                   if
+                   (os.path.isdir(self.path / folder_name) and not folder_name == "matrix_analysis")]
+
+        self.results = []
+        for folder in folders:
+            self.results.append(pd.read_csv(folder / "final_results.csv"))
+
+    def plot_given_stats(self, stats_to_plot: List[str], ax: axis):
+        for stat in stats_to_plot:
+            means = np.mean([result[stat] for result in self.results], axis=0)
+            mins = np.min([result[stat] for result in self.results], axis=0)
+            maxs = np.max([result[stat] for result in self.results], axis=0)
+            base_line, = ax.plot(means, linewidth=3, label=f"{self.prefix} {stat}")
+            ax.fill_between(self.results[0].index, maxs, mins, facecolor=base_line.get_color(), alpha=0.5)
+
+
+# configuration:
+STATS_TO_COMPARE = ["was ever sick"]
+TITLE = "matrix vs vector no policy comparison"
+TO_SHOW = False
+TO_SAVE = True
+SAVE_NAME = TITLE
+
+# simulations folders
+father_folder: Path = Path(r"C:\corona\validation\matrix-vs-vector")  # optional
```

Comment:
*parent
Did you just assume the folder's gender???

Reply:
Why hardcoded? Give as input or put in a json config file

Reply:
switched to parent.
i think this is small enough to have it's configuration inside of it.
this is suppose to be similar to the Israel comparison script.